packages/mobile/src/exchange/ExchangeConfirmationCard.tsx

```diff
@@ -39,66 +53,52 @@ class ExchangeConfirmationCard extends React.PureComponent<Props> {
         </View>
         <View style={styles.tabular}>
           <Text style={fontStyles.bodySecondary}>{t('global:celoDollars')}</Text>
-          <Text numberOfLines={1} style={[fontStyles.currency, styles.dollar]}>
-            ${newDollarBalance}
+          <Text numberOfLines={1} style={[fontStyles.body, styles.dollar]}>
+            {getMoneyDisplayValue(newDollarBalance, Tokens.DOLLAR, true)}
           </Text>
         </View>
 
         <View style={styles.tabular}>
           <Text style={fontStyles.bodySecondary}>{t('global:celoGold')}</Text>
-          <Text numberOfLines={1} style={[fontStyles.currency, styles.gold]}>
-            {newGoldBalance}
+          <Text numberOfLines={1} style={[fontStyles.body, styles.gold]}>
+            {getMoneyDisplayValue(newGoldBalance, Tokens.GOLD, true)}
           </Text>
         </View>
       </View>
     )
   }
 
-  exchangeRateToDisplay() {
-    const { leftCurrencyAmount, rightCurrencyAmount, exchangeRate } = this.props
-
-    if (exchangeRate && exchangeRate.length > 0) {
-      return new BigNumber(exchangeRate)
-    }
-
-    return rightCurrencyAmount.dividedBy(leftCurrencyAmount)
-  }
-
   render() {
-    const { newDollarBalance, newGoldBalance, leftCurrencyAmount, rightCurrencyAmount } = this.props
+    const {
+      newDollarBalance,
+      newGoldBalance,
+      makerAmount,
+      takerAmount,
+      makerToken: token,
+    } = this.props
 
-    const shouldRenderBalance = newDollarBalance.length > 0 && newGoldBalance.length > 0
     return (
       <View style={styles.container}>
         <View style={styles.exchange}>
-          <CurrencyDisplay
-            amount={leftCurrencyAmount.toString()}
-            size={36}
-            type={this.props.token}
-          />
+          <CurrencyDisplay amount={makerAmount} size={36} type={this.props.makerToken} />
           <View style={styles.arrow}>
             <RoundedArrow />
           </View>
-
-          <CurrencyDisplay
-            amount={rightCurrencyAmount.toString()}
-            size={36}
-            type={this.takerToken()}
-          />
+          <CurrencyDisplay amount={takerAmount} size={36} type={this.getTakerToken()} />
         </View>
 
         <View style={styles.title}>
-          <ExchangeRate rate={this.exchangeRateToDisplay()} makerToken={this.props.token} />
+          <ExchangeRate rate={this.getExchangeRate()} makerToken={token} />
         </View>
 
-        {shouldRenderBalance && this.renderNewBalances(newDollarBalance, newGoldBalance)}
+        {!!newDollarBalance &&
+          !!newGoldBalance &&
```

Comment:
nit: `!!` is not necessary here since we're not dealing with strings

Reply:
Fixed